mlir/lib/Dialect/Linalg/Transforms/Hoisting.cpp

```diff
@@ -43,6 +43,120 @@ using llvm::dbgs;
 using namespace mlir;
 using namespace mlir::linalg;
 
+scf::ForOp replaceWithDifferentYield(RewriterBase &rewriter, scf::ForOp loop,
```

Comment:
Would it be possible to reuse `replaceWithAdditionalYields` instead of this helper function? Something along the lines of:
- appending a new yield
- cloning the extract/broadcast outside of the loop
- updating the uses
- let the original extract/broadcast fold away

Not sure if it's worth it. Just thinking out loud...

Reply:
Oh yeah I thought about this. The main issue I have with it is that it would change the ordering of parameters (if there is more than one yielded value), which could make the pass confusing. Do you think it is preferable anyway to introducing `replaceWithDifferentYield`?